src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -168,28 +176,19 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+                Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
                 await socket.ConnectAsync(serverName, port).ConfigureAwait(false);
                 return socket;
             }
 
-            // On unix we can't use the instance Socket methods that take multiple endpoints
-
+            // On unix we can only connect using IP addresses, rather than host names.
             IPAddress[] addresses = await Dns.GetHostAddressesAsync(serverName).ConfigureAwait(false);
-            return await ConnectAsync(addresses, port).ConfigureAwait(false);
+            return await ConnectAsync(addresses, port);
         }
 
         private static async Task<Socket> ConnectAsync(IPAddress[] serverAddresses, int port)
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
```

Comment:
If, on Windows, the list of addresses are passed down, to the Native Socket APIs, then it would be beneficial to keep the Windows specific implementation around. I am yet to look at the Socket implementation on Windows.
@stephentoub Can you comment if its better to use the Socket.ConnectAsync(IPAddress [] ...) on Windows, vs spinning up parallel tasks?